packages/expo-av/ios/EXAV/EXAV.m

```diff
@@ -979,6 +1001,56 @@ - (void)sendEventWithName:(NSString *)eventName body:(NSDictionary *)body
   }
 }
 
+EX_EXPORT_METHOD_AS(getAvailableInputs,
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  NSMutableArray *inputs = [NSMutableArray new];
+  for (AVAudioSessionPortDescription *desc in [_kernelAudioSessionManagerDelegate availableInputs]){
+    [inputs addObject: @{
+      @"name": desc.portName,
+      @"type": desc.portType,
+      @"uid": desc.UID,
+    }];
+  }
+  resolve(inputs);
+}
+
+EX_EXPORT_METHOD_AS(getCurrentInput,
+                    getCurrentInput:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  AVAudioSessionPortDescription *desc = [_kernelAudioSessionManagerDelegate activeInput];
+  if (desc) {
+    resolve(@{
+      @"name": desc.portName,
+      @"type": desc.portType,
+      @"uid": desc.UID,
+    });
+  } else {
+    reject(@"E_AUDIO_GETCURRENTINPUT", @"No input port found.", nil);
+  }
+}
+
+EX_EXPORT_METHOD_AS(setInput,
+                    setInput:(NSString*)input
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+    AVAudioSessionPortDescription* preferredInput = nil;
+    for (AVAudioSessionPortDescription *desc in [_kernelAudioSessionManagerDelegate availableInputs]){
+      if ([desc.UID isEqualToString:input]) {
+          preferredInput = desc;
+      }
+    }
+    if (preferredInput != nil) {
+        [_kernelAudioSessionManagerDelegate setActiveInput:preferredInput];
+        resolve(nil);
+    } else {
+        reject(@"E_AUDIO_SETINPUT_FAIL", [NSString stringWithFormat:@"Preferred input '%@' not found!", input], nil);
+    }
```

Comment:
```suggestion
  AVAudioSessionPortDescription* preferredInput = nil;
  for (AVAudioSessionPortDescription *desc in [_kernelAudioSessionManagerDelegate availableInputs]){
    if ([desc.UID isEqualToString:input]) {
      preferredInput = desc;
    }
  }
  if (preferredInput != nil) {
    [_kernelAudioSessionManagerDelegate setActiveInput:preferredInput];
    resolve(nil);
  } else {
    reject(@"E_AUDIO_SETINPUT_FAIL", [NSString stringWithFormat:@"Preferred input '%@' not found!", input], nil);
  }
```

Reply:
thanks for all the formatting suggestions! is there a "prettier for xcode" tool that you're using?